tsl/src/compression/compression.c

```diff
@@ -2322,16 +2321,22 @@ decompress_batches(RowDecompressor *decompressor, ScanKeyData *scankeys, int num
 		if (result == TM_Updated || result == TM_Deleted)
```

Comment:
I have documented it like this:
```
/*
 * This method will:
 *  1.scan compressed chunk
 *  2.decompress the row
 *  3.delete this row from compressed chunk
 *  4.insert decompressed rows to uncompressed chunk
 *
 * Return value:
 * if all 4 steps defined above pass set chunk_status_changed to true and return true
 * if step 3 fails return false. Step 3 will fail if there are conflicting concurrent operations on same chunk.
 */
static bool
decompress_batches(RowDecompressor *decompressor, ScanKeyData *scankeys, int num_scankeys,
				   Bitmapset *null_columns, List *is_nulls, bool *chunk_status_changed)
```

Reply:
I think that's just a comment; about the operation and nothing to do with documentation

I don't think that could be accepted as a contract - it just exposes the complexity of the method...(leaving most likely the current implementation the only one which could satisfy that)

I think it should be some clear-as-day statement like:
```
Decompresses all rows satisfying the provided scankeys

It will return true if it have decompressed matching rows
```